public/src/reducers/campaign.jsx

```diff
@@ -1,11 +1,43 @@
-export const SET_CAMPAIGNS = 'SET_CAMPAIGNS';
-export const SET_CAMPAIGN_CURRENT = 'SET_CAMPAIGN_CURRENT';
-
+const scripts = [];
+const contactList = [];
 const defaultCampaigns = {
   all_campaigns: [],
   current_campaign: {}
 };
 
+export const SET_CAMPAIGN_FORM_SCRIPT_OPTIONS = 'SET_CAMPAIGN_FORM_SCRIPT_OPTIONS';
+export const SET_CAMPAIGN_FORM_CONTACT_LIST_OPTIONS = 'SET_CAMPAIGN_FORM_CONTACT_LIST_OPTIONS';
+export const SET_CAMPAIGNS = 'SET_CAMPAIGNS';
+export const SET_CAMPAIGN_CURRENT = 'SET_CAMPAIGN_CURRENT';
+
+export function scriptOptionsReducer(state = scripts, action) {
```

Comment:
Don't see why these should be separate reducers. Feel like your create campaign view should create one more reducer at most.